### PR TITLE
Validate server Properties

### PR DESCRIPTION
**Pull request by GothamsJoker**
closes Validate `server.conf` #1975

- To better validate the `server.conf` the `PropServerConfiguration` class now has a `validateProperties` method. This method compares a set of known property keys and will compare to the properties loaded by the `server.conf`. The `validateProperties` method, upon finding an incorrect key, will throw a `ServerConfigurationException` displaying the bad key or list of bad keys. 

- A test has been added to the `PropServerConfigurationTest`. The added test `testsIncorrectPropertyKeyError` will load and set invalid properties causing a `ServerConfigurationException` to be thrown. One could also simply change keys in the `server.conf` to test the added method.

- I also noticed that all of the keys in the `PropServerConfiguration` class were String literals. In this pull request, I have also added a `PropertyKey enum` containing all string literals that were used in the `PropServerConfgiuration`. The string literals passed into methods within `PropServer` are now the string values of the added enum. This will allow better forward compatibility, type safety, and easier changes in the future. Using the `PropertyKey` enum also made testing for the invalid keys simpler when using the added `getByString` method that returns an Optional of keys within.

### CODE REVIEW

**Comment by GothamsJoker** [comment]
[amunra](https://github.com/amunra) Tagging to let you know that I have worked on this. Let me know if this is a workable PR. Thank you!

File: core/src/main/java/io/questdb/PropertyKey.java [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+    LINE_TCP_UNDOCUMENTED_SYMBOL_AS_FIELD_SUPPORTED("line.tcp.undocumented.symbol.as.field.supported"),
+    LINE_TCP_UNDOCUMENTED_STRING_AS_TAG_SUPPORTED("line.tcp.undocumented.string.as.tag.supported"),
+    METRICS_ENABLED("metrics.enabled"),
+    PG_ENABLED("pg.enabled"),
+    PG_NET_CONNECTION_HINT("pg.net.connection.hint"),
+    PG_NET_BIND_TO("pg.net.bind.to"),
+    PG_NET_IDLE_TIMEOUT("pg.net.idle.timeout"),
+    PG_NET_CONNECTION_TIMEOUT("pg.net.connection.timeout"),
+    PG_NET_CONNECTION_QUEUE_TIMEOUT("pg.net.connection.queue.timeout"),
+    pg_NET_RECV_BUF_SIZE("pg.net.recv.buf.size"),
```
Comment (amunra) [inline core/src/main/java/io/questdb/PropertyKey.java:290]: `pg_NET_RECV_BUF_SIZE` -> `PG_NET_RECV_BUF_SIZE`

**Comment by ideoma** [comment]
Thanks for the PR. 
Please resolve the merge conflict and ideally please fix the minor naming inconsistency so that we can merge it ASAP.

**Comment by GothamsJoker** [comment]
Will work on this asap 😊

File: core/src/main/java/io/questdb/PropertyKey.java (hunk 1 above)
Comment (GothamsJoker) [reply core/src/main/java/io/questdb/PropertyKey.java:290]: Will fix this asap 😄

**Comment by GothamsJoker** [comment]
> Thanks for the PR. Please resolve the merge conflict and ideally please fix the minor naming inconsistency so that we can merge it ASAP.

Naming fixed and also the merge conflict!

**Pull request merged**